### PR TITLE
GIX-1777: Format age bonus as percentage

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

Consistency with the dashboard and the new neuron details UI.

In this PR: Show age and dissolve delay bonus as percentage.

# Changes

* New components DissolveDelayBonusText and AgeBonusText.
* Use new components in related ActionItems.
* Add `testId` prop to Tooltip.

# Tests

* Test file and PO for new components DissolveDelayBonusText and AgeBonusText.
* Update test cases to expect percentage.
* Use new POs in the ActioItem POs.

# Todos

Covered with the entry of the new neuron details UI.

### CODE REVIEW

**Comment by lmuntaner** [comment]
New UI:

Screenshot 2023-07-27 at 13 02 30

Screenshot 2023-07-27 at 13 03 34

**Comment by lmuntaner** [comment]
@dskloetd please review

There is an open question to product about the inconsistency with the voting power formula. https://www.notion.so/dfinityorg/Neuron-Settings-b15d6ae9be1f43aa82ae47a10121f529?pvs=4#bb49fd2e65674e88b12bc90a9b3e84a2

But that will be solved in another PR.

**Comment by lmuntaner** [comment]
> The last thing I know is that Mary was going to user test this. https://dfinity.slack.com/archives/C01SBGL2AQL/p1688651254886749?thread_ts=1688559866.557689&cid=C01SBGL2AQL Was there a conclusion from that?
> 
> I don't think we should change the numbers without changing the formula. That would make the formula wrong. Let's confirm with Mary first.

This is not yet published for users. Therefore, nobody will see the formula being wrong.

But, I'll ask @marydwyer to confirm the change and if we have a plan for the formula we can merge this.

**Comment by lmuntaner** [comment]
@dskloetd can you review this PR after confirming from product that we're using percentages? Thanks!

File: frontend/src/lib/components/neuron-detail/AgeBonusText.svelte
Comment (dskloetd) [inline frontend/src/lib/components/neuron-detail/AgeBonusText.svelte:1]: In Mary's mocks, I see the bonuses rendered as "+25%" but in your screenshots I don't see the "+".

File: frontend/src/tests/page-objects/SnsNeuronStateItemAction.page-object.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { BasePageObject } from "$tests/page-objects/base.page-object";
 import type { PageObjectElement } from "$tests/types/page-object.types";
+import { AgeBonusTextPo } from "./AgeBonusText.page-object";
 import type { ButtonPo } from "./Button.page-object";
 
 export class SnsNeuronStateItemActionPo extends BasePageObject {
```
Comment (dskloetd) [inline frontend/src/tests/page-objects/SnsNeuronStateItemAction.page-object.ts:6]: Is `SnsNeuronStateItemActionPo` identical to `NnsNeuronStateItemActionPo` except for the TID?
If yes, can you make them share the same base class and only have separate `under` functions?

Maybe for other POs in this PR as well?

Can be separate PR.

File: frontend/src/lib/components/neuron-detail/AgeBonusText.svelte [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<script lang="ts">
+  import { i18n } from "$lib/stores/i18n";
+  import { formatPercentage } from "$lib/utils/format.utils";
+  import { replacePlaceholders } from "$lib/utils/i18n.utils";
+  import Tooltip from "../ui/Tooltip.svelte";
+
+  export let ageBonus: number;
```
[the remark this answers is not in the document]
Comment (lmuntaner) [reply frontend/src/lib/components/neuron-detail/AgeBonusText.svelte:7]: sure

File: frontend/src/lib/components/neuron-detail/DissolveDelayBonusText.svelte [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<script lang="ts">
+  import { i18n } from "$lib/stores/i18n";
+  import { formatPercentage } from "$lib/utils/format.utils";
+  import { replacePlaceholders } from "$lib/utils/i18n.utils";
+  import Tooltip from "../ui/Tooltip.svelte";
+
+  export let dissolveBonus: number;
```
[the remark this answers is not in the document]
Comment (lmuntaner) [reply frontend/src/lib/components/neuron-detail/DissolveDelayBonusText.svelte:7]: sure

File: frontend/src/lib/components/neuron-detail/AgeBonusText.svelte
Comment (lmuntaner) [reply frontend/src/lib/components/neuron-detail/AgeBonusText.svelte:1]: Yes, she also mentioned it. I'll add it.

File: frontend/src/tests/page-objects/SnsNeuronStateItemAction.page-object.ts (hunk 1 above)
Comment (lmuntaner) [reply frontend/src/tests/page-objects/SnsNeuronStateItemAction.page-object.ts:6]: I would have done that if they share also the same component in the implementation.

But I haven't done that. I thought that page objects are supposed to map one to one with the components.

**Comment by lmuntaner** [comment]
@dskloetd ready for another review.

New UI:

Screenshot 2023-07-31 at 10 09 35

File: frontend/src/lib/components/neuron-detail/NnsNeuronStateItemAction.svelte [hunk 4]
```diff
@@ -42,7 +42,7 @@
   </span>
   <svelte:fragment slot="subtitle">
     {#if neuron.state === NeuronState.Locked}
-      <AgeBonusText {ageBonus} />
+      <AgeBonusText ageMultiplier={ageBonus} />
```
Comment (dskloetd) [inline frontend/src/lib/components/neuron-detail/NnsNeuronStateItemAction.svelte:45]: On line 24, you assign `ageMultiplier(...)` to `ageBonus` and then here assign `ageBonus` to `ageMultiplier`. Can it all please be consistent?

File: frontend/src/lib/components/sns-neuron-detail/SnsNeuronDissolveDelayActionItem.svelte [hunk 5]
```diff
@@ -62,7 +62,7 @@
   >
   <svelte:fragment slot="subtitle">
     {#if dissolvingTime >= minimumDelayToVoteInSeconds}
-      <DissolveDelayBonusText {dissolveBonus} />
+      <DissolveDelayBonusText dissolveMultiplier={dissolveBonus} />
```
Comment (dskloetd) [inline frontend/src/lib/components/sns-neuron-detail/SnsNeuronDissolveDelayActionItem.svelte:65]: Can this also be consistent?

File: frontend/src/tests/page-objects/SnsNeuronStateItemAction.page-object.ts (hunk 1 above)
Comment (dskloetd) [reply frontend/src/tests/page-objects/SnsNeuronStateItemAction.page-object.ts:6]: Good point! Why don't the implementations use the same component though?

**Review by dskloetd — APPROVED** [review]

File: frontend/src/lib/components/neuron-detail/NnsNeuronStateItemAction.svelte (hunk 4 above)
Comment (lmuntaner) [reply frontend/src/lib/components/neuron-detail/NnsNeuronStateItemAction.svelte:45]: `ageMultiplier` is already the name of the util. I'll use the helper inline here.

File: frontend/src/lib/components/sns-neuron-detail/SnsNeuronDissolveDelayActionItem.svelte (hunk 5 above)
Comment (lmuntaner) [reply frontend/src/lib/components/sns-neuron-detail/SnsNeuronDissolveDelayActionItem.svelte:65]: sure

File: frontend/src/lib/components/neuron-detail/NnsNeuronStateItemAction.svelte (hunk 4 above)
Comment (dskloetd) [reply frontend/src/lib/components/neuron-detail/NnsNeuronStateItemAction.svelte:45]: Not for this PR but the util should probably be called `getAgeMultiplier`, right?

File: frontend/src/tests/page-objects/SnsNeuronStateItemAction.page-object.ts (hunk 1 above)
Comment (lmuntaner) [reply frontend/src/tests/page-objects/SnsNeuronStateItemAction.page-object.ts:6]: I don't see the benefit. Most of the reused code is already in `CommonItemAction`

What maybe makes sense is to add data-tid in that shared component instead of in each usage.

I can try that in another PR.

File: frontend/src/lib/components/neuron-detail/NnsNeuronStateItemAction.svelte (hunk 4 above)
Comment (lmuntaner) [reply frontend/src/lib/components/neuron-detail/NnsNeuronStateItemAction.svelte]: We have a mix of using the prefix `getXxx` and not using it. I'd say that most of the times we don't use the prefix.

We never agreed in naming conventions.

Comment (dskloetd) [reply frontend/src/lib/components/neuron-detail/NnsNeuronStateItemAction.svelte]: Personally I think it's good to clearly distinguish function names with a verb.

Comment (lmuntaner) [reply frontend/src/lib/components/neuron-detail/NnsNeuronStateItemAction.svelte]: I agree
